### PR TITLE
Prototype to production: final material

### PR DESCRIPTION
**Pull request by riccardoporreca**
Mainly including technical step-by-step instructions and links.

### CODE REVIEW

File: from-prototype-to-production/step-by-step.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
+# MiraiLabs - From Prototype to Production
+
+## Detailed workshop steps
+
+Most of the R package development steps are carried out using the well-established R packages [devtools](https://devtools.r-lib.org/) and [usethis](https://usethis.r-lib.org/) for addressing common task of the package development workflow.
+
+
+### Create the vbzdelays package
+
+Set general author and license information
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
Set general author and license information:
```

File: from-prototype-to-production/step-by-step.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+options(
+  usethis.description = list(
+    `Authors@R` = 'person("Mirai", "Labs", email = "labs@mirai-solutions.com",
+                          role = c("aut", "cre"))',
+    License = "GPL-3"
+  )
+)
+```
+
+Create the package
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
Create the package:
```

File: from-prototype-to-production/step-by-step.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+Setup a local Git repository at the system shell via `git init` inside the package directory `vbzdelays`.
+
+From the Git pane at the top-right of RStudio (You may have to close and re-open RStudio so that the Git pane shows up):
+
+- Right-click and ignore the `.Rhistory` file
+- Add all files and commit 
+
+Create a remote repository `vbzdelays` on GitHub for your `<USERNAME>`, refusing to create initial files but copying the `.git` URL of the form `https://github.com/<USERNAME>/vbzdelays.git`.
+
+From the shell, set the remote repo for the local one created above
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
From the shell, set the remote repository for the local version of  `vbzdelays` .
```

File: from-prototype-to-production/step-by-step.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+- Add usage example for `vbzdelays::get_line_data()`
+
+Commit & push.
+
+
+### Failing Travis for broken unit-test
+
+- Refactor `compute_delay()` using `mean()`.
+- Commit & push => Travis will fail due to the broken unit test.
+- Revert via `git revert HEAD` at the system shell.
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
- Revert via `git revert HEAD` in the terminal.
```

File: from-prototype-to-production/step-by-step.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+
+### Plotting functionality  
+
+In a new source file `R/plot.R`, define an (internal) function  `barplot_by_weekday_hour` (core plot functionality) and an exported, documented `plot_delays_by_weekday_hour()` for creating the plot of a given line. Include the relevant `@import dplyr`, `@import ggplot2` tags.
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
In a new source file `R/plot.R`, define an (internal) function  `barplot_by_weekday_hour()` (core plot functionality) and an exported, documented `plot_delays_by_weekday_hour()` for creating the plot of a given line. Include the relevant `@import dplyr`, `@import ggplot2` tags.
```

File: from-prototype-to-production/step-by-step.md [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+- `browseURL(report)`
+
+Run checks locally, commit & push.
+
+
+### Website rendering functionality 
+
+- Include website index and results chapter R Markdown as `inst/site/index.Rmd` and `inst/site/01-results-by-line.Rmd`.
+- Define a new, exported and documented function `available_lines()` in `R/data.R`, used in the R Markdown above to loop over the available lines.
+- In a new source file `R/site.R`, define an exported, documented `render_render_site()`.
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
- In a new source file `R/site.R`, define an exported, documented `render_render_site()` function.
```

File: from-prototype-to-production/step-by-step.md [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+- `website <- vbzdelays::render_site(output_dir = tempfile("_site"))`
+- `browseURL(website)`
+- Update README with new example usage.
+
+Run checks locally, commit & push.
+
+
+### Setup GitHub Pages branch
+
+At the system shell, run
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
In the terminal, run:
```

File: from-prototype-to-production/step-by-step.md [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+``` bash
+git checkout --orphan gh-pages
+git rm -rf .
+echo "# Hello World" > index.md
+git add index.md
+git commit -m "Initial commit"
+git push origin gh-pages
+```
+
+You should see an Hello World website at https://USERNAME.github.io/vbzdelays.
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
You should see an "Hello World" website at https://USERNAME.github.io/vbzdelays.
```

File: from-prototype-to-production/step-by-step.md [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+Switch back to the `master` branch.
+
+
+### Continuous deployment on Travis
+
+Setup Travis CI to automatically render and deploy the website to GitHub pages upon any push to the `master` branch.
+
+See the slides for the creation of a Personal Access Token (PAT) on GitHub and definition of the secure variable `GITHUB_PAT` on Travis.
+
+Ad the following to `.travis.yaml`:
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
Add the following to `.travis.yaml`:
```

File: from-prototype-to-production/step-by-step.md [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+```
+
+Commit & push, check the progress on Travis, and then browse to the rendered website  https://USERNAME.github.io/vbzdelays.
+
+
+### First release
+
+We want to start using semantic versioning for the vbzdelays package to track future updates and corresponding changes.
+
+- Setup a change log as `NEWS.md` file via `usethis::use_news_md()`, adding relevant information about this first version of the package.
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
- Setup a changelog as `NEWS.md` file via `usethis::use_news_md()`, adding relevant information about this first version of the package.
```

File: from-prototype-to-production/step-by-step.md [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+- Define a new major version (1.0.0) via `usethis::use_version("major")`.
+- Commit & push.
+- Create release on GitHub as "v1.0.0", with title "vbzdelays 1.0.0" and content of the corresponding `NEWS.md` section.
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
- Create release on GitHub as "v1.0.0", with title "vbzdelays 1.0.0" and as content the corresponding `NEWS.md` section.
```

File: from-prototype-to-production/step-by-step.md [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
+- Pull the branch and switch to it locally (e.g. on the Git pane of RStudio)
+- _Bump_ the package revision to include .9000: `usethis::use_dev_version()`
+- Commit & push
+
+Setup branch restrictions and agile setup as GitHub project (check slides).
+
+
+### New feature 
+
+New branch for a new feature `feature/1-add-coverage-label`, created on GitHub from `develop`. Implement the feature in the local checkout of `feature/1-add-coverage-label`.
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
From `develop`, create on GitHub a novel branch for the new feature `feature/1-add-coverage-label`. Implement the feature in the local checkout of `feature/1-add-coverage-label`.
```

File: from-prototype-to-production/step-by-step.md [hunk 13]
```diff
@@ -0,0 +1,261 @@
...
+- Add a bullet about the new feature in the development section of `NEWS.md`.
+- Commit & push.
+- Create a pull request to `develop` on GitHub, get it approved and  merged.
+
+
+### New release 
+
+We want to release the new feature(s) as a minor release with version 1.1.0
+
+- Create branch `release/v1.1.0` from `develop`.
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md]:
```suggestion
- Create release branch `release/v1.1.0` from `develop`.
```

File: from-prototype-to-production/step-by-step.md [hunk 14]
```diff
@@ -0,0 +1,261 @@
...
+- Update the package version via `usethis::use_version("minor")` locally on branch `release/v1.1.0`.
+- Commit & push.
+- Create a pull request to `master` on GitHub, get it approved and merged.
+- Create release on GitHub as "v1.1.0", with title "vbzdelays 1.1.0" and content of the corresponding `NEWS.md` section.
```
Comment (fvitalini) [inline from-prototype-to-production/step-by-step.md:257]:
```suggestion
- Create release on GitHub as "v1.1.0-9000", with title "vbzdelays 1.1.0" and content of the corresponding `NEWS.md` section.
```

**Review by fvitalini — APPROVED** [review]

File: from-prototype-to-production/step-by-step.md (hunk 14 above)
Comment (riccardoporreca) [reply from-prototype-to-production/step-by-step.md:257]: This is the release, so there is no -9000

**Pull request merged**